niceml/filechecksumprocessors/filechecksumprocessor.py

```diff
@@ -0,0 +1,213 @@
+"""Module for abstract implementation of FileChecksumProcessor"""
+from abc import ABC, abstractmethod
+from collections import defaultdict
+from multiprocessing.pool import Pool
+from typing import Tuple, List, Any, Dict, Union, Optional
+
+from pydantic.utils import deep_update
+from tqdm import tqdm
+
+from niceml.utilities.fsspec.locationutils import (
+    LocationConfig,
+    open_location,
+    join_fs_path,
+)
+from niceml.utilities.ioutils import read_yaml, list_dir, write_yaml
+
+
+class FileChecksumProcessor(ABC):
+    """FileChecksumProcessor that can be used as part of a pipeline to process
+    files based on the checksum"""
+
+    # ruff: noqa: PLR0913
+    def __init__(
+        self,
+        input_location: Union[dict, LocationConfig],
+        output_location: Union[dict, LocationConfig],
+        lockfile_location: Union[dict, LocationConfig],
+        debug: bool = False,
```

Comment:
Add debug mode. If `True` no Pool should be used

Reply:
Implemented in [cc88d45](https://github.com/codecentric-oss/niceml/pull/44/commits/cc88d45d9db160ca1d28c704ec762f3c6c8024fc)